DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1

```diff
@@ -183,11 +227,23 @@ function Test-TargetResource
     try 
     {
         # Show the drive letters after mount 
+        Write-Verbose "List of PSDrives follow"
         Get-PSDrive | Write-Verbose
+        Write-Verbose "List of all partitions (selectable with PartitionNumber parameter)"
+        $mountedVHD | Get-Disk | Get-Partition | foreach{"PartitionNumber: {0}; DriveLetter: {1}; Size: {2}; Type: {3}" -f $_.PartitionNumber,$_.DriveLetter, $_.Size, $_.Type} | Write-Verbose
 
-        $mountedDrive = $mountedVHD | Get-Disk | Get-Partition | Get-Volume
-        $letterDrive  = "$($mountedDrive.DriveLetter):\"
-        Write-Verbose $letterDrive
+        $mountedPartition = $mountedVHD | Get-Disk | Get-Partition | Where{$_.PartitionNumber -like $PartitionNumber}
+        if(!$mountedPartition)
+        {
+            throw "Partition not found"
+        }
```

Comment:
Change looks good. could you resolve the conflict and make sure your changes are made on top of latest version of the files. Also,  It would be great if you could add test for this functionality (ability to choose partition number).